frontend/src/utils/server-utils.ts

```diff
@@ -302,21 +304,25 @@ export function generateIntermediateDateItemFromValidity(layer: ValidityLayer) {
   );
 
   // only calculate validity for dates that are less than 5 years old
-  const fiveYearsInMs = 5 * 365 * 24 * 60 * 60 * 1000;
+  const fiveYearsInMs = 5 * 365 * oneDayInMs;
   const earliestDate = Date.now() - fiveYearsInMs;
 
   const dateItemsWithValidity = sortedDates
     .filter(date => date > earliestDate)
-    .map(d => moment(d).set({ hour: 12, minute: 0 }))
-    .reduce((acc: DateItem[], momentDate) => {
-      // We create the start and the end date for every moment date
-      let startDate = momentDate.clone();
-      let endDate = momentDate.clone();
+    .map(d => {
+      const date = new Date(d);
+      date.setHours(12, 0, 0, 0);
```

Comment:
Should this be `setUTCHours` to ensure the user's locale has no impact? (I guess the same comment applies to all the calls of this method).